emscripten.py

```diff
@@ -1833,9 +1826,92 @@ def build_wasm(temp_files, infile, outfile, settings, DEBUG):
   return wast
 
 
-def create_metadata_wasm(metadata_raw, wast):
+def build_wasm_lld(temp_files, infile, outfile, settings, DEBUG):
+  lld = os.path.join(shared.LLVM_ROOT, 'ld.lld')
+  wasm_link_metadata = os.path.join(shared.Settings.BINARYEN_ROOT, 'bin', 'wasm-link-metadata')
+  wasm_emscripten_finalize = os.path.join(shared.Settings.BINARYEN_ROOT, 'bin', 'wasm-emscripten-finalize')
+  wasm_dis = os.path.join(shared.Settings.BINARYEN_ROOT, 'bin', 'wasm-dis')
+
+  def debug_copy(src, dst):
+    if DEBUG:
+      shutil.copyfile(src, os.path.join(shared.CANONICAL_TEMP_DIR, dst))
+      if src[-2:] == '.o' or src[-5:] == '.wasm':
+        tmp = dst + '.wast'
+        shared.check_call([wasm_dis, src, '-o', tmp])
+        shutil.copyfile(tmp, os.path.join(shared.CANONICAL_TEMP_DIR, tmp))
+
+  with temp_files.get_file('.wb.o') as temp_o:
+    backend_args = create_backend_args_wasm(infile, temp_o, settings)
+    if DEBUG:
+      logging.debug('emscript: llvm wasm backend: ' + ' '.join(backend_args))
+      t = time.time()
+    shared.check_call(backend_args)
+    if DEBUG:
+      logging.debug('  emscript: llvm wasm backend took %s seconds' % (time.time() - t))
+      t = time.time()
+      debug_copy(temp_o, 'emcc-llvm-backend-output.o')
+
+    assert shared.Settings.BINARYEN_ROOT, 'need BINARYEN_ROOT config set so we can use Binaryen tools on the backend output'
+    basename = shared.unsuffixed(outfile.name)
+    wast = basename + '.wast'
+    wasm = basename + '.wasm'
+    base_wasm = basename + '.lld.wasm'
+    meta = basename + '.json'
+    shared.check_call([wasm_link_metadata, temp_o, '-o', meta])
+    debug_copy(meta, 'lld-metadata.json')
+
+    libc_rt_lib = shared.Cache.get('wasm_libc_rt.a', wasm_rt_fail('wasm_libc_rt.a'), 'a')
+    compiler_rt_lib = shared.Cache.get('wasm_compiler_rt.a', wasm_rt_fail('wasm_compiler_rt.a'), 'a')
```

Comment:
The more "official" way of running lld is via the clang driver, ie run clang without `-c` or `-S` so it builds the LLD link-line for you. And, you can always pass additional arguments as `-Wl,--entry=main` and so on. It's the job of the Clang LLD driver to be handling libc and compiler-rt on the link-line.